cmd/signature-v4-parser.go

```diff
@@ -83,11 +83,14 @@ func parseSignature(signElement string) (string, APIErrorCode) {
 	if signFields[0] != "Signature" {
 		return "", ErrMissingSignTag
 	}
+	if signFields[1] == "" {
+		return "", ErrMissingFields
+	}
 	signature := signFields[1]
 	return signature, ErrNone
 }
 
-// Parse signed headers string.
+// Parse slice of signed headers from signed headers tag.
 func parseSignedHeaders(signedHdrElement string) ([]string, APIErrorCode) {
```

Comment:
Maybe parseSignedHeader (no 's') would be a more appropriate name, as we are only processing a single element of the header one at a time.